dill/settings.py

```diff
@@ -21,5 +23,177 @@
+    import re
+    if not option: return []  # empty value
+    option = option.strip(",\n")  # strip leading/trailing commas and empty lines
+    option = option.splitlines() if '\n' in option else re.split(r',\s+', option)
+    return [x.strip("\"'") for x in option] if strip_quotes else option
+
+def _read_filters(section, module_filters):
+    from logging.config import _resolve
+    from dill.session import EXCLUDE, INCLUDE
+    for rule_type in (EXCLUDE, INCLUDE):
+        rule = rule_type.name.lower()
+        if not any(option.lower().startswith(rule) for option in section):
+            # If no filters, let it fall back to parent module or default filters.
+            delattr(module_filters, rule)
+            continue
+        for option in (rule, '%s.names' % rule, '%s.regexes' % rule):
+            if option in section:
+                for filter in _split_option(section[option], strip_quotes=True):
+                    module_filters.add(filter, rule_type=rule_type)
+        if '%s.types' % rule in section:
+            for name in _split_option(section['%s.types' % rule]):
+                if '.' in name:
+                    filter = _resolve(name)
+                    if not isinstance(filter, type):
+                        raise TypeError("filter is not a type: %r" % filter)
+                else:
+                    filter = 'type:%s' % name
+                module_filters.add(filter, rule_type=rule_type)
+        if '%s.funcs' % rule in section:
+            for code in section['%s.funcs' % rule].strip('\n').splitlines():
+                if code.startswith('(') and code.endswith(')'):
+                    code = code[1:-1]
+                globals_ = {}
+                if not code.startswith('lambda'):
+                    name = code.partition('(')[0]
+                    _resolve(name)  # import required modules
+                    top_level = name.partition('.')[0]
+                    globals_[top_level] = __import__(top_level)
+                filter = eval(code, globals_)
+                if not callable(filter):
+                    raise TypeError("filter is not callable: %r" % filter)
+                module_filters.add(filter, rule_type=rule_type)
+
+def read_settings(filename) -> None:
+    """Read dill settings from an INI file.
+
+    Update the ``dill.settings`` dictionary with the contents of the INI file
+    ``filename``.  Accepted file sections:
+
+      - `dill`: general :py:mod:`dill` settings
+      - `dill.module`: settings for :py:func:`dill.dump_module`
+      - `filters`: default exclude/include filters for :py:func:`dill.dump_module`
+      - `filters.<module_name>`: module-specific filters for
+        :py:func:`dill.dump_module`, where `<module_name>` is the complete module
+        path in the form `module[.submodule...]`
+
+    Accepted option values for general settings:
+
+      - boolean options (case insensitive): yes, no, on, off, true, false
```

Comment:
Why not just `True` and `False`, as if it were python code?  I think anything else is pretty unnecessary and potentially confusing.

Reply:
This is the default of the `configparser` module. We can restrict the accepted values to `true` and `false` and it's case insensitive, I belive.

Reply:
Actually, the parser also accepts "0" and "1" as valid booleans. Can we just not mention the other options beyond "True" and "False"? Otherwise we can't use the default converter/validator.

Reply:
What I'm saying is that the **documentation** and **examples** should show `True` and `False`, as it will be the least confusing (and require less trying to comprehend the parsing requirements).  The parser also accepts `yes`, etc... and that's fine.  The docs should stress that it's a valid boolean that is required (i.e. `True`, `False`), and if we want to mention that other options are also valid, that's ok, but don't use the other boolean variants in the documentation/examples.